src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -93,12 +124,12 @@ export class VegaParser {
     this.tooltips = this._parseTooltips();
 
     this._setDefaultColors();
-    this._parseControlPlacement(this._config);
+    this._parseControlPlacement();
     if (this.useMap) {
       this.mapConfig = this._parseMapConfig();
-    } else if (this.spec.autosize === undefined) {
+    } else if ((this.spec as VegaSpec).autosize === undefined) {
```

Comment:
Are all of those casts really necessary? I couldn't find the code path making `this.spec` a string.

Reply:
From the constructor, we get `this.spec` as a `string`. I couldn't find usage of `this.spec` as a `string` in the code. So can I just parse the data using `hjson` right inside the constructor so that we could avoid usage of all those casts (hopefully).

Reply:
I see, it's in line https://github.com/elastic/kibana/pull/68915/files#diff-f055e57633d30c8c128f0b50547531d6R116

Can we just keep a `this.spec` as string and a `this.parsedSpec` of type `VegaSpec`? Casting in almost every single case is really ugly.